cocos/rendering/custom/index.jsb.ts

```diff
@@ -62,10 +64,17 @@ export function getCustomPipeline (name: string): PipelineBuilder {
 }
 
 export function init (device: Device, arrayBuffer: ArrayBuffer | null) {
-    if (arrayBuffer) {
-        const inflator = new zlib.Inflate(new Uint8Array(arrayBuffer));
-        const decompressed = inflator.decompress() as Uint16Array;
-        _renderModule = render.Factory.init(device, decompressed.buffer);
+    if (arrayBuffer && arrayBuffer.byteLength >= LAYOUT_HEADER_SIZE) {
+        const header = new DataView(arrayBuffer, 0, LAYOUT_HEADER_SIZE);
+        if (header.getUint32(0) === INVALID_ID) {
```

Comment:
LAYOUT_HEADER_SIZE is 8,  but uses header.getUint32() to get 4 bytes to compare?

Reply:
Only first 4 bytes are used to check compressed/uncompressed. The other 4 bytes are used as version, which is not used in engine. In editor, version is used to upgrade effect.bin.